src/main.rs

```diff
@@ -0,0 +1,206 @@
+        rest.clear();
+        let mut j = 0;
+        while j < input.len() {
+            if j != i {
+                rest.push(input[j]);
+            }
+            j += 1;
+        }
+
+        cur.push(input[i]);
+        i += 1;
+
+        let key = last_word(&cur, spaces);
+        let mut prefixes = trie.iter_prefix(key.as_bytes()).take(1);
+        match prefixes.next() {
+            Some(_) => {}
+            None => {
+                cur.pop();
+                continue;
+            }
+        }
+
+        // Try also with a longer prefix.
+        nb_results += anagrams(input_size, &rest, &cur, spaces, trie, output);
+
+        if trie.contains(&key) {
+            // Current prefix is a known word. Add a space and continue.
+            let mut new_spaces = spaces.to_vec();
+            new_spaces.push(cur.len());
+            if spaces.len() < input_size / SPACES_FACTOR {
+                nb_results += anagrams(input_size, &rest, &cur, &new_spaces, trie, output);
+            }
+        }
+
+        cur.pop();
+    }
+
+    nb_results
+}
+
+fn main() {
+    let matches = App::new("Anagramme")
+        .author("Vincent C.")
+        .arg(
+            Arg::with_name("resource-dir")
+                .short("r")
+                .long("resource-dir")
+                .value_name("DIR")
+                .required(true)
+                .help("Path to the resource directory holding dictionary files")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("language")
+                .short("l")
+                .long("language")
+                .value_name("LANG")
+                .required(true)
+                .help("Language prefix (e.g. fr for french)")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("INPUT")
+                .help("Input sentence from which anagrams are tentatively found")
+                .required(true)
+                .index(1),
+        )
+        .get_matches();
+
+    let res_dir = matches.value_of("resource-dir").unwrap();
+    let input = matches.value_of("INPUT").unwrap().to_lowercase();
+    let lang = matches.value_of("language").unwrap().to_lowercase();
+
+    let txtfile = format!("{}.txt", lang);
+
+    let raw_dict_fr = get_raw_dict(txtfile, res_dir);
+    let mut trie = PatriciaSet::new();
+
+    let mut i = 0;
+    let mut start = 0;
+    while i < raw_dict_fr.len() {
+        if raw_dict_fr[i] == b'\n' {
+            let u8_word = &raw_dict_fr[start..i];
+            let word = str::from_utf8(u8_word).unwrap();
+            let ascii_word: String = word.nfd().filter(char::is_ascii).collect();
+            trie.insert(ascii_word.to_lowercase());
+            i += 1;
+            start = i;
+            continue;
+        }
+        i += 1;
+    }
+
+    let input_vec = input.replace(" ", "").as_bytes().to_vec();
+    let mut outputs = HashMap::new();
+    anagrams(
+        input_vec.len(),
```

Comment:
```suggestion
```

Reply:
This is needed to short-circuit the algorithm when too many spaces would be added.